src/vtbackend/Functions.h

```diff
@@ -618,10 +623,82 @@ inline auto const& functions() noexcept
     return funcs;
 }
 
+class SupportedSequences
+{
+  public:
+    SupportedSequences()
+    {
+        for (auto [index, seq]: crispy::indexed(functions()))
+        {
+            _supportedSequences[index] = seq;
+        }
+        crispy::sort(
+            _supportedSequences,
+            [](FunctionDefinition const& a, FunctionDefinition const& b) constexpr { return compare(a, b); });
+        _lastIndex = _supportedSequences.size();
+    }
+
+    [[nodiscard]] gsl::span<FunctionDefinition const> getAllSequences() const noexcept
+    {
+        return gsl::span<FunctionDefinition const>{_supportedSequences.cbegin(), _supportedSequences.cend()};
+    }
+
+    [[nodiscard]] gsl::span<FunctionDefinition const> getSupportedSequencesAtCurrentVTLevel() const noexcept
+    {
+        return gsl::span<FunctionDefinition const>{_supportedSequences.cbegin(),
+                                                   _supportedSequences.cbegin() + _lastIndex};
+    }
+
+    void updateVTLevel(VTType vt) noexcept
+    {
+        auto itr = std::partition(
+            _supportedSequences.begin(),
+            _supportedSequences.end(),
+            [vt](const FunctionDefinition& value) noexcept { return value.conformanceLevel <= vt; });
+
+        _lastIndex = std::distance(_supportedSequences.begin(), itr);
+        gsl::span<FunctionDefinition> availableDefinition{_supportedSequences.begin(), _lastIndex};
+        crispy::sort(
+            availableDefinition,
+            [](FunctionDefinition const& a, FunctionDefinition const& b) constexpr { return compare(a, b); });
+    }
+
+    void disableSequence(FunctionDefinition seq) noexcept
+    {
+        auto seqIter = std::find(_supportedSequences.begin(), _supportedSequences.begin() + _lastIndex, seq);
+        if (seqIter != _supportedSequences.begin() + _lastIndex)
+        {
+            std::rotate(seqIter, seqIter + 1, _supportedSequences.end());
+            --_lastIndex;
+        }
+    }
+
+    void enableSequence(FunctionDefinition seq) noexcept
+    {
+        auto* seqIter = std::find(_supportedSequences.begin() + _lastIndex, _supportedSequences.end(), seq);
+        if (seqIter != _supportedSequences.end())
+        {
+            // Maybe could be done better since rest of the data is sorted
+            std::iter_swap(_supportedSequences.begin() + _lastIndex, seqIter);
```

Comment:
things like `_supportedSequences.begin()` as well as `_supportedSequences.begin() + _lastIndex` are written so often, it actually makes sense to provide helper members for them:

```cpp
constexpr auto begin() noexcept { return _supportedSequences.begin(); }
constexpr auto end() noexcept { return std::next(_supportedSequences.begin(), _lastIndex); }
```

And then use these everywhere where we'd have manually written it like this.